app/containers/MessageActions/index.js

```diff
@@ -193,8 +193,22 @@ const MessageActions = React.memo(forwardRef(({
 	const handleStar = async(message) => {
 		logEvent(message.starred ? events.ROOM_MSG_ACTION_UNSTAR : events.ROOM_MSG_ACTION_STAR);
 		try {
-			await RocketChat.toggleStarMessage(message.id, message.starred);
-			EventEmitter.emit(LISTENER, { message: message.starred ? I18n.t('Message_unstarred') : I18n.t('Message_starred') });
+			const db = database.active;
+			await db.action(async() => {
+				await message.update((m) => {
+					m.starred = !m.starred;
+				});
+			});
+			const result = await RocketChat.toggleStarMessage(message.id, !message.starred);
+			if (result.success === true) {
+				EventEmitter.emit(LISTENER, { message: !message.starred ? I18n.t('Message_unstarred') : I18n.t('Message_starred') });
+			} else {
```

Comment:
@diegolmello Please check here.

If API call is a success then Show Toast Message ` Message Starred/Unstarred`.
Else if API call errored out then again change the value in local database, hence again going back to old (and right) state.
Thanks. :)